tests/test_versions.bats

```diff
@@ -0,0 +1,32 @@
+# -*- mode: bash-ts -*-
+
+function setup {
+  bats_load_library bats-support
+  bats_load_library bats-assert
+
+  load "$DIRENV_STDLIB"
+  load "$DIRENVRC"
+}
+
+function _require_version_with_valid_versions { # @test
+  # args: cmd version minimum_required
+  run _require_version "test-cmd" "2.5" "2.4"
+  assert_success
+  run _require_version "test-cmd" "2.5" "2.4.1"
+  assert_success
+  run _require_version "test-cmd" "2.4.1" "2.4"
+  assert_success
+  run _require_version "test-cmd" "2.4" "2.4.1"
+  assert_failure
+  run _require_version "test-cmd" "2.31pre20250712_b1245123" "2.4"
+  assert_success
+}
+
+function _require_cmd_version_with_valid_versions { # @test
+  run _require_cmd_version "bash" "1.0"
```

Comment:
I had trouble with the functions -as-commands approach that @fidgetingbits was using in his, so I just added bash with ridiculous version requirements. If someone knows what might be the cause of failures with the approach to this taken in #593, I'm happy to revert to that approach.

Reply:
Yes - the error I was seeing was that the "function-as-command" approach ended up with empty output. I'm not entirely sure why either.